.github/workflows/build.yml

```diff
@@ -78,6 +78,34 @@ jobs:
           token: f421b687-4dc2-4387-ac3d-dc3b2528af57
           fail_ci_if_error: true
 
+  check_commits:
+    runs-on: ubuntu-latest
+    env:
+      TOOLCHAIN: stable
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install Rust ${{ env.TOOLCHAIN }} toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ env.TOOLCHAIN }}
+          override: true
+          profile: minimal
+      - name: Fetch full tree and rebase on upstream
+        run: |
+          git remote add upstream https://github.com/rust-bitcoin/rust-lightning
+          git fetch upstream
+          git rebase upstream/main
+      - name: For each commit, run cargo check (including in fuzz)
+        run: |
+          for COMMITHASH in `git log --format=format:%H upstream/main...HEAD`; do
```

Comment:
Can this be a shell script that devs can run locally, too? Makes it easier to test as well.

Reply:
A bit more complicated to scan the remotes, but, done!